core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -401,15 +403,16 @@ private[spark] object JettyUtils extends Logging {
       uri.append(rest)
     }
 
-    val rewrittenURI = URI.create(uri.toString())
+    val rewrittenURI = URI.create(uri.toString()).normalize()
```

Comment:
For this one, I am not very confident to decode the URI, which may break the `authority` part of the URI.